doc/kdump/SONiC-kdump.md

```diff
@@ -0,0 +1,425 @@
+
+7. Core files are compressed by the *makedumpfile* program to occupy less storage space on disk.
+
+8. Much information (such as free pages) in a vmcore (kernel core) is not useful. *makedumpfile* excludes unnecessary pages, such as:
+ - pages filled with zeroes,
+ - cache pages without private flag (non-private cache);
+ - cache pages with private flag (private cache);
+ - user process data pages;
+ - free pages.
+We let makedumpfile to save only in-use kernel pages to reduce the corefile.
+
+8. File permissions of generated core files generated are set appropriately so that only a root user can access them.
+
+9. When too many core files are generated and take up more than the allocated disk space, the oldest core files are automatically deleted to make room for the newly created ones.
+
+### SONiC Code Changes
+
+Current SONiC code has basic kernel support to generate a kernel core file, but user-space support is missing. To add full kdump support, the following changes are proposed:
+
+ - Install *kdump-tools* and *makedumpfile* as part of SONiC host O/S.
+Changes to be done in *sonic-buildimage/build_debian.sh*
+ - kdump-tools package is modified:
+   - to allow the creation of the initramfs image for the capture kernel on first boot, since the standard kdump-tools Debian package does not allow creating the initramfs image from a fakeroot environment during image build time.
+   - the default maximum number of stored core files is set to 3.
+   - kernel core dump files are prefixed by *kdump-*
+   - kdump-tools serive is enabled or disabled depending of administrative state (as stored in ConfigDB)
+ - kdump is configured using the SONiC utility *config*.
+ - kdump status and configuration can be shown using the SONiC utility *show*.
+ - kdump is disabled by default.
+ - if kdump is enabled and a new SONiC image is installed using the *sonic_installer* command, kdump will be enabled in the new installed image.
+
+### Configuration commands:
+
+For SONiC switches, the following CLI commands are provided to manage core files.
+
+#### config kdump <enable|disable> [-y] [-n]
+
+Enable or disable kernel core dump functionality.
+
+Administrative state of kdump is stored in ConfigDB.
+
+The variable *USE_KDUMP* in the file /etc/default/kdump-tools is set to 0 to disable kdump, and set to 1 to enable kdump.
+
+Since this command might require changing the kernel parameters to specify the amount of memory reserved for the capture kernel (the kernel parameters which are exported through */proc/cmdline*), a reboot is necessary. By default, the command displays a message showing that kdump functionality will be either enabled or disabled following the next reboot, unless the optional parameter *-n* has been provided. The optional parameter *-y* allows to automatically perform the reboot without asking for a confirmation. In this case, the command "config save -y" will be automatically performed prior to the reboot.
+
```

Comment:
user can choose to do warm reboot or fast reboot, recommend to do not integrate reboot into this config tool.